src/libraries/System.Private.CoreLib/src/System/Numerics/BitOperations.cs

```diff
@@ -708,5 +711,168 @@ public static nuint RotateRight(nuint value, int offset)
             return (nuint)RotateRight((uint)value, offset);
 #endif
         }
+
+        /// <summary>
+        /// Calculates the CRC (Cyclic redundancy check) checksum.
+        ///
+        /// Uses following Hardware Intrinsics, if supported:
+        /// * uint32_t __crc32b (uint32_t a, uint8_t b)
+        /// * uint32_t __crc32cb (uint32_t a, uint8_t b)
+        /// </summary>
+        /// <param name="crc">The base value to calculate checksum on</param>
+        /// <param name="data">The checksum data</param>
+        /// <returns>The CRC-checksum</returns>
+        [CLSCompliant(false)]
+        public static uint Crc32C(uint crc, byte data)
+        {
+            if (Sse42.IsSupported)
+            {
+                // uint32_t __crc32b (uint32_t a, uint8_t b)
+                return Sse42.Crc32(crc, data);
+            }
+            if (Crc32.IsSupported)
+            {
+                // uint32_t __crc32cb (uint32_t a, uint8_t b)
+                return Crc32.ComputeCrc32C(crc, data);
+            }
+
+
+            // Software fallback
+            int tableIndex = (int)((crc ^ data) & 0xFF);
```

Comment:
It's worth using `GetArrayDataReference` and `Unsafe.Add`ing the index here as is done in the other methods.

BTW, I'm assuming you got this reference implementation from a C sample?  `& 0xFF` is necessary in C to isolate the low byte, because C doesn't guarantee the upper bits will be cleared.  Most C compilers will optimize this to a zero extension (e.g. `movzx`) where possible.  JIT will take your code literally and emit the `and`.  A cast to `byte` (as used in the other methods) is sufficient to clear the upper bits because .NET guarantees this where C doesn't.